src/project.cpp

```diff
@@ -1563,94 +1547,89 @@ void Project::readInitialFacingDirections() {
-    QString filepath = root + "/include/constants/weather.h";
+    QString filename = "include/constants/weather.h";
     QStringList prefixes = (QStringList() << "COORD_EVENT_WEATHER_");
-    readCDefinesSorted(filepath, prefixes, coordEventWeatherNames);
+    readCDefinesSorted(filename, prefixes, coordEventWeatherNames);
 }
 
 void Project::readSecretBaseIds() {
-    QString filepath = root + "/include/constants/secret_bases.h";
+    QString filename = "include/constants/secret_bases.h";
     QStringList prefixes = (QStringList() << "SECRET_BASE_[A-Za-z0-9_]*_[0-9]+");
-    readCDefinesSorted(filepath, prefixes, secretBaseIds);
+    readCDefinesSorted(filename, prefixes, secretBaseIds);
 }
 
 void Project::readBgEventFacingDirections() {
-    QString filepath = root + "/include/constants/bg_event_constants.h";
+    QString filename = "include/constants/bg_event_constants.h";
     QStringList prefixes = (QStringList() << "BG_EVENT_PLAYER_FACING_");
-    readCDefinesSorted(filepath, prefixes, bgEventFacingDirections);
+    readCDefinesSorted(filename, prefixes, bgEventFacingDirections);
 }
 
 void Project::readMetatileBehaviors() {
     this->metatileBehaviorMap.clear();
     this->metatileBehaviorMapInverse.clear();
-    QString filepath = root + "/include/constants/metatile_behaviors.h";
-    QString text = readTextFile(filepath);
-    if (!text.isNull()) {
-        QStringList prefixes = (QStringList() << "MB_");
-        this->metatileBehaviorMap = readCDefines(text, prefixes);
-        for (QString defineName : this->metatileBehaviorMap.keys()) {
-            this->metatileBehaviorMapInverse.insert(this->metatileBehaviorMap[defineName], defineName);
-        }
-    } else {
-        logError(QString("Failed to read C defines file: '%1'").arg(filepath));
+    QString filename = "include/constants/metatile_behaviors.h";
+
+    QStringList prefixes = (QStringList() << "MB_");
+    this->metatileBehaviorMap = readCDefines(filename, prefixes);
+    for (QString defineName : this->metatileBehaviorMap.keys()) {
+        this->metatileBehaviorMapInverse.insert(this->metatileBehaviorMap[defineName], defineName);
     }
 }
 
-void Project::readCDefinesSorted(QString filepath, QStringList prefixes, QStringList* definesToSet) {
-    QString text = readTextFile(filepath);
-    if (!text.isNull()) {
-        QMap<QString, int> defines = readCDefines(text, prefixes);
-
-        // The defines should to be sorted by their underlying value, not alphabetically.
-        // Reverse the map and read out the resulting keys in order.
-        QMultiMap<int, QString> definesInverse;
-        for (QString defineName : defines.keys()) {
-            definesInverse.insert(defines[defineName], defineName);
-        }
-        *definesToSet = definesInverse.values();
-    } else {
-        logError(QString("Failed to read C defines file: '%1'").arg(filepath));
+void Project::readCDefinesSorted(QString filename, QStringList prefixes, QStringList* definesToSet) {
```

Comment:
I feel like the `readCDefines` functions should live in `parseUtil.cpp`.  Agree?

Reply:
yes